src/desktop/credentials/DesktopNativeCredentialsFacade.ts

```diff
@@ -96,20 +93,35 @@ export class DesktopNativeCredentialsFacade implements NativeCredentialsFacade {
 		const storedAppPassSaltB64 = await this.conf.getVar(DesktopConfigKey.appPassSalt)
 		if (storedAppPassSaltB64 == null) return null
 		const commonNativeFacade = await this.getCurrentCommonNativeFacade()
-		const pw = await commonNativeFacade.promptForPassword(this.lang.get("credentialsEncryptionModeAppPassword_label"))
+		const pw = await this.tryWhileSaltNotChanged(commonNativeFacade.promptForPassword(this.lang.get("credentialsEncryptionModeAppPassword_label")))
 		const salt = base64ToUint8Array(storedAppPassSaltB64)
 		return generateKeyFromPassphraseArgon2id(await this.argon2idFacade, pw, salt)
 	}
 
 	private async enrollForAppPass(): Promise<Aes256Key> {
 		const newSalt = this.crypto.randomBytes(KEY_LENGTH_BYTES_AES_256)
 		const commonNativeFacade = await this.getCurrentCommonNativeFacade()
-		const newPw = await commonNativeFacade.promptForNewPassword(this.lang.get("credentialsEncryptionModeAppPassword_label"), null)
+		const newPw = await this.tryWhileSaltNotChanged(
+			commonNativeFacade.promptForNewPassword(this.lang.get("credentialsEncryptionModeAppPassword_label"), null),
+		)
 		const newAppPassSaltB64 = uint8ArrayToBase64(newSalt)
 		await this.conf.setVar(DesktopConfigKey.appPassSalt, newAppPassSaltB64)
 		return generateKeyFromPassphraseArgon2id(await this.argon2idFacade, newPw, newSalt)
 	}
 
+	private async tryWhileSaltNotChanged(pwPromise: Promise<string>): Promise<string> {
+		const commonNativeFacade = await this.getCurrentCommonNativeFacade()
+		return resolveChecked<string>(
```

Comment:
I am not sure why we need it, it is a confusing abstraction to me, it took me multiple minutes to understand what it does, even with a comment.

Why is it not

```ts
try {
 return await Promise.race(passwordPromise, cancellationPromise)
}  catch (e) {
  if (e instanceof CancelledError) {
      commonNativeFacade.showAlertDialog("retry_action")
  }

  throw e
```

Is it because you want to wait for the password to be done before showing the dialog?

Reply:
It would be more clear inline but I understand that you want to test it.